src/components/data/EntitiesContent.js

```diff
@@ -511,11 +511,14 @@ const EntitiesContent = ({
       deletingEntities && h(EntityDeleter, {
         onDismiss: () => setDeletingEntities(false),
         onSuccess: () => {
-          setDeletingEntities(false)
           setSelectedEntities({})
           setRefreshKey(_.add(1))
           Ajax().Metrics.captureEvent(Events.workspaceDataDelete, extractWorkspaceDetails(workspace.workspace))
-          loadMetadata()
+
+          // Avoid React warning about state update on unmounted component.
+          // Loading metadata will cause EntitiesComponent to be unmounted and replaced with a spinner while
+          // metadata is loading. Defer loading metadata until after the confirmation modal is dismissed.
+          Utils.onNextTick(loadMetadata)
```

Comment:
As discussed, think about whether the two state transitions (and associated renders) in a row will be visually surprising to the user. If so, this is a reasonable approach. If not, you'll need to make some kind of change to the pattern.

Reply:
I took the need for using `Utils.onNextTick` here and in #2950, along with the need for adding the `dismissOnError` option, to be evidence that this was the wrong abstraction. Refactored DeleteConfirmationModal so that it only calls `onConfirm` and dismissing the modal and managing busy state are left to the parent component.